### PR TITLE
update to support the new build system

### PR DESCRIPTION
**Pull request by boddumanohar**
This Pull request adds support the new build system.
Tested with the branch https://github.com/rianquinn/hypervisor/tree/rfc_602

PS: This is my first PR in c++ language. So please be gentle :wink: 

Thanks,
Manohar.

### CODE REVIEW

File: src/vcpu_factory_vpid.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// headers for vcpu
```
Comment (rianquinn) [inline src/vcpu_factory_vpid.cpp]: remove comment

File: src/vcpu_factory_vpid.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// headers for vcpu 
+#include <bfvmm/vcpu/vcpu_factory.h>
+#include <bfvmm/vcpu/arch/intel_x64/vcpu.h>
+
+// headers for vmcs
```
Comment (rianquinn) [inline src/vcpu_factory_vpid.cpp]: remove dead code

File: src/vcpu_factory_vpid.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+//#include <arch/intel_x64/vmcs/16bit_control_fields.h>
+//#include <arch/intel_x64/vmcs/32bit_control_fields.h>
+//#include <bfvmm/hve/arch/intel_x64/vmcs/vmcs.h>
+//#include <bfdebug.h>
+
+//#include <arch/intel_x64/exit_handler/exit_handler.h>
```
Comment (rianquinn) [inline src/vcpu_factory_vpid.cpp]: remove comment

File: src/vcpu_factory_vpid.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+namespace example
+{
+namespace intel_x64
+{
+
+class vcpu : public bfvmm::intel_x64::vcpu
+{
+public:
+
+// create vcpu
```
Comment (rianquinn) [inline src/vcpu_factory_vpid.cpp]: remove comment

File: src/vcpu_factory_vpid.cpp [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    vcpu(vcpuid::type id) :
+        bfvmm::intel_x64::vcpu{id}
+    {
+	static uint16_t vpid = 1;
+	virtual_processor_identifier::set(vpid++);
+        secondary_processor_based_vm_execution_controls::enable_vpid::enable();
```
Comment (rianquinn) [inline src/vcpu_factory_vpid.cpp]: minor style issues. Should look like
```
static uint16_t vpid = 1;

virtual_processor_identifier::set(vpid++);
secondary_processor_based_vm_execution_controls::enable_vpid::enable();
```

File: src/vcpu_factory_vpid.cpp [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    }
+
+    ~vcpu() override
```
Comment (rianquinn) [inline src/vcpu_factory_vpid.cpp]: should be:

```
~vcpu() override = default;
```

**Review by rianquinn — CHANGES_REQUESTED** [review]

**Comment by rianquinn** [comment]
Overall this looks good. Please fix the above issues and we should be good a merge

**Comment by rianquinn** [comment]
Also, rebase and "fixup" you commit into a single commit

**Comment by tklengyel** [comment]
@rianquinn: github now lets you squash PRs into a single commit at merge time! It also gets rid of the "merge" commit.

File: src/vcpu_factory_vpid.cpp [hunk 7]
```diff
@@ -0,0 +1,42 @@
+#include <bfvmm/vcpu/vcpu_factory.h>
+#include <bfvmm/vcpu/arch/intel_x64/vcpu.h>
+using namespace ::intel_x64::vmcs;
```
Comment (rianquinn) [inline src/vcpu_factory_vpid.cpp]: space between include and using

File: src/vcpu_factory_vpid.cpp [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+WEAK_SYM std::unique_ptr<vcpu>
+vcpu_factory::make_vcpu(vcpuid::type vcpuid, bfobject *obj)
+{
+    bfignored(obj);
+    return std::make_unique<example::intel_x64::vcpu>(vcpuid);
+}
+
+}
+
+
```
Comment (rianquinn) [inline src/vcpu_factory_vpid.cpp]: remove extra newline

**Review by rianquinn — CHANGES_REQUESTED** [review]

**Review by rianquinn — APPROVED** [review]

**Pull request merged**